src/components/RestaurantList.tsx

```diff
@@ -1,127 +1,105 @@
+        initSelectOption();
+      });
+    } catch (e) {
+      console.log(e);
     }
+  };
 
-    const filteredRestaurants = this.state.restaurants.filter(
-      (restaurant) => restaurant.category === this.state.category
-    );
-    this.setState({ filteredRestaurants: filteredRestaurants });
-  }
+  const initSelectOption = () => {
+    setCategory(CategoryKind.all);
+    setOrder(OrderKind.name);
 
-  componentDidUpdate(prevProps: object, prevStates: RestaurantListStateType) {
-    if (prevStates.category !== this.state.category) {
-      this.filterRestaurants();
-      window.scrollTo(0, 0);
-    }
-    if (prevStates.order !== this.state.order || prevStates.filteredRestaurants !== this.state.filteredRestaurants) {
-      this.sortRestaurants();
-      window.scrollTo(0, 0);
+    handleClickEvent();
+  };
+
+  const filterRestaurants = () => {
+    if (category === CategoryKind.all) {
+      setFilteredRestaurants(localRestaurants);
+      return;
     }
-  }
 
-  render() {
+    const filteredRestaurantsData = localRestaurants.filter((restaurant) => restaurant.category === category);
+    setFilteredRestaurants(filteredRestaurantsData);
+  };
+
+  useEffect(() => {
+    setLocalRestaurants(data);
+  }, []);
+
+  useEffect(() => {
+    setFilteredRestaurants(localRestaurants);
+    handleClickEvent();
+  }, [localRestaurants]);
+
+  useEffect(() => {
+    filterRestaurants();
+    window.scrollTo(0, 0);
+  }, [category]);
+
+  const filteredRestaurantElement = () => {
     return (
       <>
-        <SelectBoxContainer>
-          <SelectBox
-            selectType={SelectKind.category}
-            options={Object.values(CATEGORY_NAME)}
-            onChange={(event: React.ChangeEvent<HTMLSelectElement>) => {
-              this.setState({ category: event.target.value });
-            }}
-          />
-          <SelectBox
-            selectType={SelectKind.order}
-            options={Object.values(ORDER_KEY)}
-            onChange={(event: React.ChangeEvent<HTMLSelectElement>) => {
-              this.setState({ order: event.target.value });
-            }}
-          />
-        </SelectBoxContainer>
-        <RestaurantListWrapper>
-          {this.state.filteredRestaurants.map((restaurant: RestaurantItemType, index: number) => {
-            return (
-              <RestaurantItem
-                key={index}
-                category={restaurant.category}
-                name={restaurant.name}
-                distance={restaurant.distance}
-                description={restaurant.description}
-                link={restaurant.link}
-              />
-            );
+        {filteredRestaurants
+          .sort((a, b) => {
+            if (order === OrderKind.name) return a.name > b.name ? 1 : -1;
+            return a.distance > b.distance ? 1 : -1;
+          })
+          .map((restaurant: RestaurantItemType) => {
+            return <RestaurantItem key={restaurant.id} {...restaurant} />;
```

Comment:
요런 계산은 return() 전에 계산하고 오면 좋을 것 같아요!

Reply:
아니면 그냥 바로 활용하는 것도 좋을 것 같네요!
```suggestion
      const filteredRestaurantElement = filteredRestaurants
          .sort((a, b) => {
            if (order === OrderKind.name) return a.name > b.name ? 1 : -1;
            return a.distance > b.distance ? 1 : -1;
          })
          .map((restaurant: RestaurantItemType) => 
            <RestaurantItem key={restaurant.id} {...restaurant} />
          );
```